### PR TITLE
Watch out for moves with null profiles

### PR DESCRIPTION
**Pull request by mafonso**
### Jira link

P4-1788

### What?

I have added/removed/altered:

- [ ] Handle moves with no profiles.

### Why?

I am doing this because:

- In staging we have some data quality issues. So this is to protect against those cases

### Have you? (optional)

- [ ] Updated API docs if necessary	
- [ ] Added environment variables to [deployment repository](https://github.com/ministryofjustice/hmpps-book-secure-move-api-deploy)

### Deployment risks (optional)

- Includes destructive/migratory actions which may effect critical offender data
- Changes an api that is used in production

### CODE REVIEW

File: lib/tasks/journeys_report.rake [hunk 1]
```diff
@@ -67,7 +67,7 @@ namespace :journeys do
           date: move.date,
           from: move.from_location.nomis_agency_id,
           to: move.to_location&.nomis_agency_id,
-          person_id: move.profile.person_id,
+          person_id: move.profile&.person_id,
```
Comment (martyn-w) [inline lib/tasks/journeys_report.rake:70]: you'll need to check for move.profile.person throughout this file (e.g. gender, date_of_birth etc) and update accordingly

Comment (mafonso) [reply lib/tasks/journeys_report.rake:70]: Included all the references to `profile` in `moves`

File: lib/tasks/journeys_report.rake [hunk 2]
```diff
@@ -67,10 +67,10 @@ namespace :journeys do
...
           from: move.from_location.nomis_agency_id,
           to: move.to_location&.nomis_agency_id,
-          person_id: move.profile.person_id,
-          gender: move.person.gender&.key,
+          person_id: move.profile&.person_id,
+          gender: move.person.gender&.keys,
           dob: move.person.date_of_birth,
-          age: ((move.date.to_date - move.profile.person.date_of_birth.to_date) / 365.25).to_i,
+          age: ((move.date.to_date - move.person.date_of_birth.to_date) / 365.25).to_i,
           events: move.move_events.default_order.map do |move_event|
```
Comment (mafonso) [inline lib/tasks/journeys_report.rake:74]: Some rebase needed,  `gender` and ` date_of_birth` we get from `persons`.

File: config/spring.rb [hunk 3]
```diff
@@ -4,3 +4,11 @@
   tmp/restart.txt
   tmp/caching-dev.txt
 ].each { |path| Spring.watch(path) }
+Spring.after_fork do
```
Comment (martyn-w) [inline config/spring.rb:7]: oops - I don't think this debugger stuff is intended for here

Comment (mafonso) [reply config/spring.rb:7]: Oops, it ended up getting  pushed.

**Pull request merged**